pkg/util/iptables/save_restore.go

```diff
@@ -61,7 +61,11 @@ func GetChainLines(table Table, save []byte) map[Chain][]byte {
 		} else if line[0] == ':' && len(line) > 1 {
 			// We assume that the <line> contains space - chain lines have 3 fields,
 			// space delimited. If there is no space, this line will panic.
-			chain := Chain(line[1:bytes.Index(line, spaceBytes)])
+			spaceIndex := bytes.Index(line, spaceBytes)
+			if spaceIndex == -1 {
+				panic(fmt.Sprintf("Unexpected chain line in iptables-save output: %v", string(line)))
```

Comment:
Is a panic preferable to introducing an error return?

Reply:
The original code panicked, but with a non-useful message. This change keeps the panic, but improves the message, so it's easier to see what the problem with the iptables-save output is. 

Returning an error would require a bigger change to all the callers. Plus, it's impossible to know how to handle the error.